client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -218,20 +215,18 @@ public synchronized SchemaMetadata getLatestSchemaMetadata(String subject)
   @Override
   public synchronized int getVersion(String subject, Schema schema)
       throws IOException, RestClientException {
-    Map<Schema, Integer> schemaVersionMap;
+    IdentityLRUCache<Schema, Integer> schemaVersionMap;
     if (versionCache.containsKey(subject)) {
       schemaVersionMap = versionCache.get(subject);
     } else {
-      schemaVersionMap = new IdentityHashMap();
+      schemaVersionMap = new IdentityLRUCache<>(identityMapCapacity);
       versionCache.put(subject, schemaVersionMap);
     }
 
-    if (schemaVersionMap.containsKey(schema)) {
-      return schemaVersionMap.get(schema);
+    Integer id = schemaVersionMap.get(schema);
+    if (id != null) {
+      return id;
```

Comment:
Same comment on `return id` outside the if/else as above. Also a bit confusing here, returning id or version. Use `id` in both branches?